### PR TITLE
Implement the T-distribution via the F-distribution

### PR DESCRIPTION
**Pull request by andreasnoack**
Today I realized that

1. I already implemented a pure Julia version of the F-distribution functions
2. We can just implement the `T(ν)`-distribution via the `F(1,ν)`-distribution 

Hence this PR. This is an alternative to #147

### CODE REVIEW

File: src/distrs/tdist.jl [hunk 1]
```diff
@@ -22,3 +8,67 @@ function tdistlogpdf(ν::T, x::T) where {T<:Real}
...
+tdistcdf(ν::Real, x::Real) = tdistcdf(map(float, promote(ν, x))...)
+
+tdistccdf(ν::Real, x::Real) = tdistcdf(ν, -x)
+
+function tdistlogcdf(ν::T, x::T) where T<:Real
+    if isinf(ν)
+        return normlogcdf(x)
+    elseif x < 0
+        ret = fdistlogccdf(one(ν), ν, x*x)
+        return ret - log(2*one(ret))
```
Comment (ararslan) [inline src/distrs/tdist.jl:30]: FWIW I believe this should also work as expected and avoids needing to actually compute the log:
```suggestion
        return ret - logtwo
```

File: src/distrs/tdist.jl [hunk 2]
```diff
@@ -22,3 +8,67 @@ function tdistlogpdf(ν::T, x::T) where {T<:Real}
     νp12 = (ν + 1) / 2
     return loggamma(νp12) - (logπ + log(ν)) / 2 - loggamma(ν / 2) - νp12 * log1p(x^2 / ν)
 end
+
+function tdistcdf(ν::T, x::T) where T<:Real
+    if isinf(ν)
+        return normcdf(x)
+    elseif x < 0
+        return fdistccdf(one(ν), ν, x*x)/2
```
Comment (devmotion) [inline src/distrs/tdist.jl:16]: I would assume nowadays the compiler is smart enough to compile optimized code if we use
```suggestion
        return fdistccdf(one(ν), ν, x^2)/2
```
instead?

**Comment by andreasnoack** [comment]
Thanks. I've pushed a commit that addresses the comments.

File: src/distrs/tdist.jl [hunk 3]
```diff
@@ -22,3 +8,67 @@ function tdistlogpdf(ν::T, x::T) where {T<:Real}
...
+
+tdistlogccdf(ν::Real, x::Real) = tdistlogcdf(ν, -x)
+
+function tdistinvcdf(ν::T, p::T) where T<:Real
+    if isinf(ν)
+        return norminvcdf(p)
+    elseif p < 0.5
+        return -sqrt(fdistinvccdf(one(ν), ν, 2*p))
+    else
+        sqrt(fdistinvccdf(one(ν), ν, 2*(1 - p)))
```
Comment (ararslan) [inline src/distrs/tdist.jl:45]:
```suggestion
        return sqrt(fdistinvccdf(one(ν), ν, 2*(1 - p)))
```

File: src/distrs/tdist.jl [hunk 4]
```diff
@@ -22,3 +8,67 @@ function tdistlogpdf(ν::T, x::T) where {T<:Real}
...
+    end
+end
+tdistinvcdf(ν::Real, p::Real) = tdistinvcdf(map(float, promote(ν, p))...)
+
+tdistinvccdf(ν::Real, p::Real) = -tdistinvcdf(ν, p)
+
+if VERSION < v"1.7.0-beta1"
```
Comment (ararslan) [inline src/distrs/tdist.jl:52]:
```suggestion
if VERSION < v"1.7.0-DEV.1172"
```
As an aside, perhaps this method should be added to Compat.

File: src/distrs/tdist.jl [hunk 5]
```diff
@@ -22,3 +8,67 @@ function tdistlogpdf(ν::T, x::T) where {T<:Real}
...
+end
+_expm1(x::Number) = expm1(x)
+
+function tdistinvlogcdf(ν::T, logp::T) where T<:Real
+    if isinf(ν)
+        return norminvlogcdf(logp)
+    elseif logp < -log(2)
+        return -sqrt(fdistinvlogccdf(one(ν), ν, logp + logtwo))
+    else
+        sqrt(fdistinvccdf(one(ν), ν, -2*_expm1(logp)))
```
Comment (ararslan) [inline src/distrs/tdist.jl:69]:
```suggestion
        return sqrt(fdistinvccdf(one(ν), ν, -2*_expm1(logp)))
```

**Review by ararslan — APPROVED** [review]
Very minor comments around consistency but otherwise looks great. Math checks out AFAICT and the tests are happy. Thanks!

**Pull request merged**